### PR TITLE
fix: Clear input after adding a repository to insight page

### PR DESCRIPTION
**Pull request by shubhamchasing**
## Description
Passed the value prop to Search component 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
fixes [#2474](https://github.com/open-sauced/app/issues/2474) 

## Mobile & Desktop Screenshots/Recordings
https://github.com/open-sauced/app/assets/106425076/4e1a914c-63c1-49ab-849e-d3f9a35b5e35

## Steps to QA

1. Navigate to insight page -> click New Insight button
2. Add repo 
3. Now Input field is empty

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?
giphy

### CODE REVIEW

**Comment by shubhamchasing** [comment]
@nickytonline can you please help me here, I am new to this process

**Comment by BekahHW** [comment]
Hey @shubhamchasing! Thanks for the PR. Take a look at [the directions](https://github.com/open-sauced/app/pull/2497#pullrequestreview-1831684203) for how you need to format your PR to get the tests to run. 

Another note: You'll need to update your branch with the latest from beta before this can be merged.

**Review by BekahHW — CHANGES_REQUESTED** [review]
When I try to select from the combobox, it doesn't work on Chrome. 

https://github.com/open-sauced/app/assets/34313413/4c13c8a7-4a7a-4fed-b76b-6e260fdcf0ff

**Comment by shubhamchasing** [comment]
Hey @BekahHW 
To find this [bug](https://github.com/open-sauced/app/pull/2497#pullrequestreview-1836719912)  I checked alot of things and I think it occured 
after [feat: flow for creating a new workspace](https://github.com/open-sauced/app/pull/2461) PR link for  [(preview)](https://deploy-preview-2461--oss-insights.netlify.app/hub/insights/new)
in the [feat: search suggestions can be JSX or a string now](https://github.com/open-sauced/app/pull/2489) PR link for  [(preview)](https://deploy-preview-2489--oss-insights.netlify.app/hub/insights/new) 

I also noticed some random behaviour, like I am able to add some repo and some not and it also changes everytime

**Comment by nickytonline** [comment]
> Hey @BekahHW To find this [bug](https://github.com/open-sauced/app/pull/2497#pullrequestreview-1836719912) I checked alot of things and I think it occured after [feat: flow for creating a new workspace](https://github.com/open-sauced/app/pull/2461) PR, link for [(preview)](https://deploy-preview-2461--oss-insights.netlify.app/hub/insights/new) in the [feat: search suggestions can be JSX or a string now](https://github.com/open-sauced/app/pull/2489) PR, link for [(preview)](https://deploy-preview-2489--oss-insights.netlify.app/hub/insights/new)
> 
> I also noticed some random behaviour, like I am able to add some repo and some not and it also changes everytime

Let me take a peek at this @shubhamchasing as the issues you mention are some recent work I did. It's very possible I introduced this issue. 😅

File: components/organisms/InsightPage/InsightPage.tsx
Comment (nickytonline) [inline components/organisms/InsightPage/InsightPage.tsx:555]: Most likely this is still not clearing as reported by @BekahHW  because the state is not being reset, i.e. `setRepoSearchTerm("")`. I see there are calls to this already, but maybe one is missing for this new use case of the search being a [controlled component](https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components) now.

Comment (shubhamchasing) [reply components/organisms/InsightPage/InsightPage.tsx:555]: You @nickytonline  are right search component is now a controlled component.

And I think `loadAndAddRepo` is handling the use case which is `setRepoSearchTerm("")` . Please let me know if I misunderstood something.

``` 
 const loadAndAddRepo = async (repoToAdd: string, isAddedFromCart = false) => {
    setAddRepoError(RepoLookupError.Initial);


    const hasRepo = repos.find((repo) => `${repo.full_name}` === repoToAdd);


    if (hasRepo) {
      return;
    }


    const { apiPaths, isValidUrl } = generateRepoParts(repoToAdd);


    if (!isValidUrl) {
      setAddRepoError(RepoLookupError.Invalid);
      return;
    }


    const { repoFullName } = apiPaths;


    setAddRepoLoading({ repoName: repoToAdd, isAddedFromCart, isLoading: true });
    try {
      const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/repos/${repoFullName}`);
      setAddRepoLoading({ repoName: repoToAdd, isAddedFromCart, isLoading: false });


      if (response.ok) {
        const addedRepo = (await response.json()) as DbRepo;


        setRepos((repos) => {
          return [...repos, addedRepo];
        });
        setAddRepoError(RepoLookupError.Initial);
        setRepoSearchTerm("");
      } else {
        const publicRepoResponse = await fetch(`https://api.github.com/repos/${repoFullName}`);


        if (publicRepoResponse.ok) {
          const publicRepo = await publicRepoResponse.json();


          // create a stub repo to send to API
          const addedRepo = {
            id: publicRepo.id,
            full_name: publicRepo.full_name,
          } as DbRepo;


          setRepos((repos) => {
            return [...repos, addedRepo];
          });
          setAddRepoError(RepoLookupError.Initial);
          setRepoSearchTerm("");
        } else {
          setAddRepoError(RepoLookupError.Invalid);
        }
      }
    } catch {
      setAddRepoLoading({ repoName: repoToAdd, isAddedFromCart, isLoading: false });
      setAddRepoError(RepoLookupError.Error);
    }
  };


  const handleAddRepository = async () => {
    await loadAndAddRepo(repoSearchTerm);
  };
```

[Link to code](https://github.com/open-sauced/app/blob/75992f53dc389709f26de88f3ad3a335cdc1ac51/components/organisms/InsightPage/InsightPage.tsx#L271C2-L331C5)

**Comment by shubhamchasing** [comment]
> When I try to select from the combobox, it doesn't work on Chrome.
> 
>  Screen.Recording.2024-01-22.at.10.15.08.AM.mov

- This issue was newly introduced which I mentioned earlier in the [comment](https://github.com/open-sauced/app/pull/2497#issuecomment-1906171904).

- And now the issue has been resolved in the [#2502](https://github.com/open-sauced/app/pull/2502) by using `currentTarget`

``` 
const { suggestion } = (event.currentTarget as HTMLElement).dataset;
   ```

[Screencast from 29-01-24 10:05:21 PM IST.webm](https://github.com/open-sauced/app/assets/106425076/cd6c7e6e-6ad5-4805-85c2-9786ea6a4a4e)

File: components/organisms/InsightPage/InsightPage.tsx
Comment (shubhamchasing) [reply components/organisms/InsightPage/InsightPage.tsx:555]: @nickytonline waiting for your response 😅

**Comment by shubhamchasing** [comment]
Hey guys @BekahHW @nickytonline @jpmcb, please look into this PR when you have time :)

**Review by jpmcb — APPROVED** [review]
Looks good to me: apologies for the long wait on this. We've been heads down shipping some new features and things.

Nick is OOO this week but I can leave it to others in engineering to comment if we want to bring this in. Thanks again so much for your contribution and patience!

**Review by BekahHW — APPROVED** [review]

**Pull request merged**
